apps/api/src/domain/models/term/commands/add-audio-for-term/audio-added-for-term.event-handler.spec.ts

```diff
@@ -0,0 +1,143 @@
+    type: 'TERM_TRANSLATED',
+});
+
+const audioItemId = buildDummyUuid(1);
+
+const audioAddedForTerm = termTranslated.andThen<AudioAddedForDigitalTextPage>({
+    type: 'AUDIO_ADDED_FOR_DIGITAL_TEXT_PAGE',
+    payload: {
+        audioItemId,
+    },
+});
+
+const [creationEvent, translationEvent, audioAddedEvent] =
+    audioAddedForTerm.as(termCompositeIdentifier);
+
+const initialView = TermViewModel.fromTermCreated(creationEvent as TermCreated).apply(
+    translationEvent
+);
+
+describe('AudioAddedForTermEventHandler.handle', () => {
+    let testQueryRepository: ITermQueryRepository;
+
+    let databaseProvider: ArangoDatabaseProvider;
+
+    let arangoDatabaseForCollection: ArangoDatabaseForCollection<
+        IDetailQueryResult<ITermViewModel>
+    >;
+
+    let app: INestApplication;
+
+    let audioAddedForTermEventHandler: AudioAddedForTermEventHandler;
+
+    beforeAll(async () => {
+        const moduleRef = await Test.createTestingModule({
+            imports: [PersistenceModule.forRootAsync()],
+        })
+            .overrideProvider(ConfigService)
+            .useValue(
+                buildMockConfigService(
+                    {
+                        ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                        // TODO this shouldn't be necessary
+                        ARANGO_DB_HOST_PORT: 8551,
+                    },
+                    buildConfigFilePath(Environment.test)
+                )
+            )
+            .compile();
+
+        await moduleRef.init();
+
+        app = moduleRef.createNestApplication();
+
+        const connectionProvider = app.get(ArangoConnectionProvider);
+
+        databaseProvider = new ArangoDatabaseProvider(connectionProvider);
+
+        arangoDatabaseForCollection = databaseProvider.getDatabaseForCollection('term__VIEWS');
+
+        testQueryRepository = new ArangoTermQueryRepository(
+            connectionProvider,
+            app.get(AUDIO_QUERY_REPOSITORY_TOKEN),
+            new ConsoleCoscradCliLogger()
+        );
+
+        audioAddedForTermEventHandler = new AudioAddedForTermEventHandler(testQueryRepository);
+    });
+
+    afterAll(async () => {
+        databaseProvider.close();
+    });
+
+    beforeEach(async () => {
+        arangoDatabaseForCollection.clear();
+
+        /**
+         * We attempted to use "handle" on a creation event for the test
+         * setup, but it failed due to an apparent race condition.
+         *
+         * We should investigate this further.
+         */
+        await testQueryRepository.create(initialView);
+    });
+
+    describe(`when there is an existing term`, () => {
+        it(`should update the database appropriately`, async () => {
+            await audioAddedForTermEventHandler.handle(audioAddedEvent as AudioAddedForTerm);
+
+            const result = await testQueryRepository.fetchById(termId);
+
+            // Do we want to return instances?
```

Comment:
No. Remove this. We want the query layer to be highly optimized with minimal abstractions.